rts/lib/headlessStubs/CMakeLists.txt

```diff
@@ -5,9 +5,8 @@ SET(headlessStubsSources
 		"sdlstub.c"
 	)
 
-FIND_PACKAGE(SDL2 REQUIRED)
-INCLUDE_DIRECTORIES(${SDL2_INCLUDE_DIR})
+find_package(SDL2 MODULE REQUIRED)
 
 ADD_LIBRARY(headlessStubs STATIC EXCLUDE_FROM_ALL ${headlessStubsSources})
+target_link_libraries(headlessStubs PUBLIC SDL2::SDL2)
```

Comment:
headlessStubs should not be linking sdl2. I've also not noticed this in the previous PR only now, headless shouldn't be also linking GLEW. The same applies to dedicated.

So, 5e604820512b9677a627210ec12537b42b2f47de started unnecessarily linking GLEW to headless and dedicated, this library adds sdl2 to the mix.

Reply:
Yeah I noticed that too -- it actually still needs headers -- cmake 3.27 introduces a "include only" link -- maybe we could do that?

Reply:
Could you point me at the docs? Can't find it.

We can update build environment to cmake 3.27. Maximal version we can easy upgrade to is 3.28.4, because since 3.29 cmake requires python>=3.7 which the ancient ubuntu 18 doesn't have.

Reply:
This guy: `COMPILE_ONLY` generator expression https://discourse.cmake.org/t/add-only-library-headers-during-target-link-libraries/2973/15

Reply:
Okey https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#compile-context, looks perfect!